### ISSUE TITLE
Add support for recovering a sandcats domain whose key has been misplaced

### ISSUE THREAD

**Issue by paulproteus** [sandstorm-io/sandcats#49]
I imagine the flow for this is:

In installer, if you type `help` (like how `none` is special-cased), then the install script will say:

```
If you lost your sandcats domain, and you need it back, you can type its
name here.

If you need other help, type help again.
What Sandcats name do you need help with? asheesh.sandcats.io
OK! We've emailed you a token to enter into this installer that will
let you use that domain on this new Sandstorm install.
Please enter the token: acfd0ed6fc274e9d3e272d1bae6279df
OK! We'll use asheesh.sandcats.io for the install.
```

**Comment by paulproteus** [comment]
@phildini got stuck in this state, so this is probably a show-stopper before 1.0.

**Comment by kentonv** [comment]
If you type a name that's already taken, we could present the option to recover that name then. I think that's what most people are likely to try.

**Comment by phildini** [comment]
I think I actually _did_ try to put the same name in twice, in the hope it would give me some sort of recovery option.

**Comment by paulproteus** [comment]
+1, will expose this in the error messages

**Comment by paulproteus** [comment]
I believe this is closed now, as of #59.
